integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -57,28 +57,60 @@
       outputDatasetSupplier;
   private final List<QueryPlanVisitor<LogicalPlan, OpenLineage.InputDataset>> inputDatasetSupplier;
 
+  private AtomicBoolean started = new AtomicBoolean(false);
+  private AtomicBoolean finished = new AtomicBoolean(false);
+
   public SparkSQLExecutionContext(
       long executionId,
       OpenLineageContext sparkContext,
+      QueryExecution queryExecution,
       List<QueryPlanVisitor<LogicalPlan, OpenLineage.OutputDataset>> outputDatasetSupplier,
       List<QueryPlanVisitor<LogicalPlan, OpenLineage.InputDataset>> inputDatasetSupplier) {
     this.executionId = executionId;
     this.sparkContext = sparkContext;
-    this.queryExecution = SQLExecution.getQueryExecution(executionId);
+    this.queryExecution = queryExecution;
     this.outputDatasetSupplier = outputDatasetSupplier;
     this.inputDatasetSupplier = inputDatasetSupplier;
   }
 
-  public void start(SparkListenerSQLExecutionStart startEvent) {}
+  public void start(SparkListenerSQLExecutionStart startEvent) {
+    log.debug("SparkListenerSQLExecutionStart - executionId: " + startEvent.executionId());
```

Comment:
As a general rule, we should stick with `log.debug("blah blah {}", argument)` rather than using string concatenation

Reply:
Fixed.